src/middleware/builtin.ts

```diff
@@ -110,8 +119,17 @@ export function matchConstraints(
           return;
         }
       } else {
-        if ((tempMatches = constraints.action_id.exec(payload.action_id)) !== null) {
-          matches.concat(tempMatches);
+        matches = [];
+        while ((tempMatches = constraints.action_id.exec(payload.action_id)) !== null) {
+          matches.push(tempMatches);
+          // Avoids an infinite loop if /g isn't at the end of the RegExp
+          if (!constraints.action_id.global) {
+            break;
+          }
+        }
+
+        if (matches.length !== 0) {
+          context['actionId'] = matches;
```

Comment:
```suggestion
          context['matches'] = matches;
```

i think that's what you actually meant.